mlir/test/Dialect/Tosa/constant-reshape-fold.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: mlir-opt --split-input-file --tosa-layerwise-constant-fold %s | FileCheck %s
+// RUN: mlir-opt --split-input-file --tosa-layerwise-constant-fold="fold-splat-or-single-use-only=0" %s | FileCheck %s --check-prefix CHECK-MULTI
```

Comment:
Is `CHECK-MULTI` used frequently already? Otherwise I would suggest to go for `always fold` or the like. When I read the `reshape_multi_user_splat` test I was confused at first because it has multiple users and is folded even without `CHECK-MULTI`, but that is because it is splat. So since the multi-user thing is not really the only criterion, but it is forcing the folding to happen, I would make it more about "always fold, without using any heuristic".

Reply:
I changed this to CHECK-ALWAYS and added comments explaining the behavior